### PR TITLE
#164798159 Setup React Router

### PR DESCRIPTION
**Pull request by dannylwe**
#### What does this PR do?
Setup routing for the react project

#### Description of Task to be completed?
Enables the user to route between different components

#### How should this be manually tested?
fetch branch `git fetch ch-setup-react-router-164798159`
checkout branch `git checkout ch-setup-react-router-164798159`
run `npm install`
run `npm start`
You can test this by going to the "/login" endpoint from the landing page

#### What are the relevant pivotal tracker stories?
[#164798159](https://www.pivotaltracker.com/story/show/164798159)

#### Screenshots (if appropriate)
image
image

### CODE REVIEW

File: src/components/Landing.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const Login = () => (
```
Comment (ezrogha) [inline src/components/Landing.js]: Please let the file name be the same as the default export component

Comment (adesege) [inline src/components/Landing.js]: The component name does not match the name of the file.

**Review by adesege — CHANGES_REQUESTED** [review]

File: src/components/App.js [hunk 2]
```diff
@@ -1,8 +1,14 @@
 import React from 'react';
+import { Route, BrowserRouter as Router } from 'react-router-dom';
+import Login from './Login';
+import Landing from './Landing';
 
 const App = () => (
   <div>
-    <h1>Authors Haven</h1>
+    <Router>
```
Comment (ezrogha) [inline src/components/App.js]: I suggest routes be in a file Routes.js, as the App.js may be our root for redux

**Review by ezrogha — CHANGES_REQUESTED** [review]
Well done @dannylwe

File: src/components/Landing.js (hunk 1 above)
Comment (dannylwe) [reply src/components/Landing.js]: @adesege worked on feedback

Comment (dannylwe) [reply src/components/Landing.js]: @ezrogha worked on feedback

File: src/components/App.js (hunk 2 above)
Comment (dannylwe) [reply src/components/App.js]: @ezrogha implemented suggestion

**Review by felixkiryowa — APPROVED** [review]

File: src/components/Landing.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const Landing = () => (
```
Comment (ezrogha) [inline src/components/Landing.js:3]: Hey @dannylwe, well done with the feedback.
Please change this file/component name to something like Home.js, 
Landing is a bit misleading

**Review by ezrogha — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 .idea/
 node_modules/
 coverage/
+package-lock.json
```
Comment (adesege) [inline .gitignore]: Do not ignore this file. It should always be committed

File: src/components/App.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
+import Routes from './Routes';
 
 const App = () => (
   <div>
```
Comment (adesege) [inline src/components/App.js]: You can use `React.Fragment` here so as not to render the extra div in the dom.

File: src/components/Routes.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Route, BrowserRouter as Router } from 'react-router-dom';
+import Login from './Login';
+import Landing from './Landing';
+
+const Routes = () => (
+  <div>
```
Comment (adesege) [inline src/components/Routes.js]: Use React.Fragment here instead

**Review by adesege — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (dannylwe) [reply .gitignore]: @adesege worked on feedback

File: src/components/App.js (hunk 5 above)
Comment (dannylwe) [reply src/components/App.js]: @adesege worked on feedback

File: src/components/Routes.js (hunk 6 above)
Comment (dannylwe) [reply src/components/Routes.js]: @adesege worked on feedback

File: src/components/Routes.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Route, BrowserRouter as Router } from 'react-router-dom';
+import Login from './Login';
+import Landing from './Landing';
+
+const Routes = () => (
+  <>
+    <Router>
+      <Route exact path="/" component={Landing} />
+      <Route path="/login" component={Login} />
```
Comment (adesege) [inline src/components/Routes.js:10]: There's no test file for Login component.

File: src/tests/__snapshots__/App.test.js.snap [hunk 8]
```diff
@@ -1,9 +1,7 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`App should render without crashing 1`] = `
-<div>
-  <h1>
-    Authors Haven
-  </h1>
-</div>
+<h1>
```
Comment (adesege) [inline src/tests/__snapshots__/App.test.js.snap:4]: Update the test file to reflect the name of the component you are testing for.

**Review by adesege — CHANGES_REQUESTED** [review]

**Review by ezrogha — APPROVED** [review]
LGTM

File: src/components/Routes.js (hunk 7 above)
Comment (dannylwe) [reply src/components/Routes.js:10]: On it...

File: src/tests/__snapshots__/App.test.js.snap (hunk 8 above)
Comment (dannylwe) [reply src/tests/__snapshots__/App.test.js.snap:4]: @adesege the tests have been updated. The App component renders the Route component. The default route is the Landing page component. So I think this test is fine.

File: src/components/Routes.js (hunk 7 above)
Comment (dannylwe) [reply src/components/Routes.js:10]: @adesege worked on feedback

File: src/tests/Login.test.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Login from '../components/Login';
+
+describe('App', () => {
```
Comment (adesege) [inline src/tests/Login.test.js]: The title does not reflect the name of the component you are testing for

File: src/tests/Landing.test.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Landing from '../components/Landing';
+
+describe('App', () => {
```
Comment (adesege) [inline src/tests/Landing.test.js]: The title does not reflect the name of the component you are testing for

**Review by adesege — CHANGES_REQUESTED** [review]

**Review by adesege — APPROVED** [review]

**Pull request merged**